### DISCUSSION TITLE
Building GCC inside VS Code

### DISCUSSION THREAD

**Discussion by MathiasMagnus** [microsoft/vscode-makefile-tools#560] [Q&A]
I have tried to configure and build GCC inside VS Code with the extension but ultimately couldn't.

IIUC GCC `configure` script would behave as a `makefile.PreConfigureScript`, creating the `Makefile` for the build, and as such it has to run at least once. However [GCC's configure](https://gcc.gnu.org/install/configure.html) throws a hard error when trying to build in-source, and I can't seem to find a way to tell it to place the makefiles other than cwd and the extension doesn't let me configure the cwd while invoking the pre-configure script.

Has anyone seen success building GCC with this extension?

**Comment by itodirel** [comment]
Are you able to successfully build GCC in the command line? GCC is a notorious difficult project to configure. This seems like a good test, let us know if you ran into any issues, and we are open to contributions in the extension.

**Reply by MathiasMagnus** [reply]
On the comand-line yes, it's a `<path_to_gcc_root>/configure <your_config_args>` and then `make -j$(nproc)`. (Running the tests and deciding whether the tests pass is another issue, but I won't go into that cause I couldn't say nice things.) So the configure/build process is fairly straightforward on the command-line.

Being a CMake user primarily, the bootstrap builds both CMake and LLVM do don't mix well with configuring IntelliSense, so that part I did not expect to work flawlessly. I did wanted to kick off the configure/build/test workflow from the GUI, but ultimately just went with the command-line.

**Reply by gcampbell-msft** [reply]
@MathiasMagnus Looking at this again, what is the hard error that GCC's configure hits? I'm not sure I understand why GCC's configure would fail in our extension but not on the command-line, maybe I'm missing some understanding about how to build gcc?
